📖 Add OpenStack to External Cloud Providers in quick-start.md (#10061)
kubernetes-sigs/cluster-api: docs/book/src/user/quick-start.md

For more information, see the [CAPZ book](https://capz.sigs.k8s.io/topics/addons.html).

{{#/tab }}
{{#tab OpenStack}}

> Before deploying the Calico CNI, configure the `cloud.conf` file for integration with your OpenStack environment:

Reviewer: ```suggestion
Before deploying the OpenStack external cloud provider, configure the `cloud.conf` file for integration with your OpenStack environment:
```

```bash
cat > cloud.conf <<EOF
[Global]
auth-url=<your_auth_url>
application-credential-id=<your_credential_id>
application-credential-secret=<your_credential_secret>
region=<your_region>
domain-name=<your_domain_name>
EOF
```

Next, create a Kubernetes secret using this configuration to securely store your cloud environment details.
You can create this secret for example with:

```bash

> kubectl create secret generic capi-quickstart-cloud-conf --from-file=cloud.conf

Reviewer: The name and namespace has to match what is used in the manifests
```suggestion
kubectl -n kube-system create secret generic cloud-config --from-file=cloud.conf
```
Ref https://github.com/kubernetes/cloud-provider-openstack/blob/64b813046f25b41aa4295a0a51726bcf25e92bc7/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml#L78-L80

```

> 

Reviewer: We should add here how to actually install it. Maybe something like this:

Now you are ready to deploy the external cloud provider!

```bash
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-roles.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-role-bindings.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml
```

Alternatively, refer to the [helm chart](https://github.com/kubernetes/cloud-provider-openstack/tree/master/charts/openstack-cloud-controller-manager).

For more information, see the [CAPZ book](https://capz.sigs.k8s.io/topics/addons.html).

{{#/tab }}
{{#tab OpenStack}}

Before deploying the OpenStack external cloud provider, configure the `cloud.conf` file for integration with your OpenStack environment:

```bash
cat > cloud.conf <<EOF
[Global]
auth-url=<your_auth_url>
application-credential-id=<your_credential_id>
application-credential-secret=<your_credential_secret>
region=<your_region>
domain-name=<your_domain_name>

> EOF

Reviewer 2: Could you link to the docs explaining the configuration too? They're here: https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md#config-openstack-cloud-controller-manager

Author: Yes thanks, it is done now.

```

Next, create a Kubernetes secret using this configuration to securely store your cloud environment details.
You can create this secret for example with:

```bash
kubectl create secret generic capi-quickstart-cloud-conf --from-file=cloud.conf
```

Now, you are ready to deploy the external cloud provider!

```bash
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-roles.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-role-bindings.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml
```

> Alternatively, refer to the [helm chart](https://github.com/kubernetes/cloud-provider-openstack/tree/master/charts/openstack-cloud-controller-manager)

Reviewer 3: ```suggestion
Alternatively, refer to the [helm chart](https://github.com/kubernetes/cloud-provider-openstack/tree/master/charts/openstack-cloud-controller-manager).
```

Sorry, one more nit :-)

As of that lgtm 🎉

{{#tab OpenStack}}

Before deploying the OpenStack external cloud provider, configure the `cloud.conf` file for integration with your OpenStack environment:

```bash
cat > cloud.conf <<EOF
[Global]
auth-url=<your_auth_url>
application-credential-id=<your_credential_id>
application-credential-secret=<your_credential_secret>
region=<your_region>
domain-name=<your_domain_name>
EOF
```

Next, create a Kubernetes secret using this configuration to securely store your cloud environment details.
You can create this secret for example with:

```bash
kubectl -n kube-system create secret generic cloud-config --from-file=cloud.conf
```

Now, you are ready to deploy the external cloud provider!

```bash
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-roles.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-role-bindings.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml
```

> Alternatively, refer to the [helm chart] (https://github.com/kubernetes/cloud-provider-openstack/tree/master/charts/openstack-cloud-controller-manager)

Reviewer 3: ```suggestion
Alternatively, refer to the [helm chart](https://github.com/kubernetes/cloud-provider-openstack/tree/master/charts/openstack-cloud-controller-manager)
```

Otherwise the link is not a link.
